cartography/intel/aws/cloudtrail_management_events.py

```diff
@@ -0,0 +1,553 @@
+import json
+import logging
+from datetime import datetime
+from datetime import timedelta
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+import boto3
+import neo4j
+
+from cartography.client.core.tx import load_matchlinks
+from cartography.graph.job import GraphJob
+from cartography.intel.aws.ec2.util import get_botocore_config
+from cartography.models.aws.cloudtrail.management_events import AssumedRoleMatchLink
+from cartography.util import aws_handle_regions
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+@aws_handle_regions
+def get_cloudtrail_events(
+    boto3_session: boto3.Session, region: str, lookback_hours: int = 24
+) -> List[Dict[str, Any]]:
+    """
+    Fetch CloudTrail management events from the specified time period.
+
+    :type boto3_session: boto3.Session
+    :param boto3_session: The boto3 session to use for API calls
+    :type region: str
+    :param region: The AWS region to fetch events from
+    :type lookback_hours: int
+    :param lookback_hours: Number of hours back to retrieve events from
+    :rtype: List[Dict[str, Any]]
+    :return: List of CloudTrail events
+    """
+    client = boto3_session.client(
+        "cloudtrail", region_name=region, config=get_botocore_config()
+    )
+
+    # Calculate time range
+    end_time = datetime.utcnow()
+    start_time = end_time - timedelta(hours=lookback_hours)
+
+    logger.info(
+        f"Fetching CloudTrail management events for region '{region}' "
+        f"from {start_time} to {end_time} ({lookback_hours} hours)"
+    )
+
+    events = []
+
+    try:
+        # Focus on STS events for role assumptions
+        paginator = client.get_paginator("lookup_events")
+
+        page_iterator = paginator.paginate(
```

Comment:
Is there a way to filter the event name from the API in the `get` function so that we don't need to do it in the transform function later? This would let us save on memory